diff -ruN -x '.*' cld2-org/cld2_api.cpp cld2-patched/cld2_api.cpp
--- cld2-org/cld2_api.cpp	1970-01-01 01:00:00.000000000 +0100
+++ cld2-patched/cld2_api.cpp	2026-09-13 14:49:49.563098436 +0200
@@ -0,0 +1,194 @@
+#include "cld2_api.h"
+
+#include "compact_lang_det.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+using namespace CLD2;
+
+static const char* lang_str(Language lang) {
+    switch (lang) {
+        case ENGLISH:
+            return "en";
+        case DANISH:
+            return "da";
+        case DUTCH:
+            return "nl";
+        case FINNISH:
+            return "fi";
+        case FRENCH:
+            return "fr";
+        case GERMAN:
+            return "de";
+        case HEBREW:
+            return "iw";
+        case ITALIAN:
+            return "it";
+        case JAPANESE:
+            return "ja";
+        case KOREAN:
+            return "ko";
+        case NORWEGIAN:
+            return "no";
+        case POLISH:
+            return "pl";
+        case PORTUGUESE:
+            return "pt";
+        case RUSSIAN:
+            return "ru";
+        case SPANISH:
+            return "es";
+        case SWEDISH:
+            return "sv";
+        case CHINESE:
+            return "zh";
+        case CZECH:
+            return "cs";
+        case GREEK:
+            return "el";
+        case ICELANDIC:
+            return "is";
+        case LATVIAN:
+            return "lv";
+        case LITHUANIAN:
+            return "lt";
+        case ROMANIAN:
+            return "ro";
+        case HUNGARIAN:
+            return "hu";
+        case ESTONIAN:
+            return "et";
+        case TG_UNKNOWN_LANGUAGE:
+            return "";
+        case UNKNOWN_LANGUAGE:
+            return "";
+        case BULGARIAN:
+            return "bg";
+        case CROATIAN:
+            return "hr";
+        case SERBIAN:
+            return "sr";
+        case IRISH:
+            return "ga";
+        case GALICIAN:
+            return "gl";
+        case TAGALOG:
+            return "tl";
+        case TURKISH:
+            return "tr";
+        case UKRAINIAN:
+            return "uk";
+        case HINDI:
+            return "hi";
+        case MACEDONIAN:
+            return "mk";
+        case BENGALI:
+            return "bn";
+        case INDONESIAN:
+            return "id";
+        case LATIN:
+            return "la";
+        case MALAY:
+            return "ms";
+        case MALAYALAM:
+            return "ml";
+        case WELSH:
+            return "cy";
+        case NEPALI:
+            return "ne";
+        case TELUGU:
+            return "te";
+        case ALBANIAN:
+            return "sq";
+        case TAMIL:
+            return "ta";
+        case BELARUSIAN:
+            return "be";
+        case JAVANESE:
+            return "jw";
+        case OCCITAN:
+            return "oc";
+        case URDU:
+            return "ur";
+        case BIHARI:
+            return "bh";
+        case GUJARATI:
+            return "gu";
+        case THAI:
+            return "th";
+        case ARABIC:
+            return "ar";
+        case CATALAN:
+            return "ca";
+        case ESPERANTO:
+            return "eo";
+        case BASQUE:
+            return "eu";
+        case INTERLINGUA:
+            return "ia";
+        case KANNADA:
+            return "kn";
+        case PUNJABI:
+            return "pa";
+        case SCOTS_GAELIC:
+            return "gd";
+        case SWAHILI:
+            return "sw";
+        case SLOVENIAN:
+            return "sl";
+        case MARATHI:
+            return "mr";
+        case MALTESE:
+            return "mt";
+        case VIETNAMESE:
+            return "vi";
+        case FRISIAN:
+            return "fy";
+        case SLOVAK:
+            return "sk";
+        case CHINESE_T:
+            return "zh-Hant";
+        case FAROESE:
+            return "fo";
+        case SUNDANESE:
+            return "su";
+        case UZBEK:
+            return "uz";
+        case AMHARIC:
+            return "am";
+        case AZERBAIJANI:
+            return "az";
+        case GEORGIAN:
+            return "ka";
+        case TIGRINYA:
+            return "ti";
+        case PERSIAN:
+            return "fa";
+        case BOSNIAN:
+            return "bs";
+        case SINHALESE:
+            return "si";
+        case NORWEGIAN_N:
+            return "nn";
+    }
+    return "";
+}
+
+const char* cld2_detect_language(const char* buffer, int buffer_length) {
+    CLD2::CLDHints cld_hints{};
+    CLD2::Language language3[3];
+    int percent3[3];
+    double normalized_score3[3];
+    bool is_reliable = false;
+    int text_bytes = 0;
+
+    auto lang = CLD2::ExtDetectLanguageSummary(
+        buffer, buffer_length, /*is_plain_text=*/true, &cld_hints, /*flags=*/0,
+        language3, percent3, normalized_score3, /*resultchunkvector=*/nullptr,
+        &text_bytes, &is_reliable);
+
+    return lang_str(lang);
+}
diff -ruN -x '.*' cld2-org/cld2_api.h cld2-patched/cld2_api.h
--- cld2-org/cld2_api.h	1970-01-01 01:00:00.000000000 +0100
+++ cld2-patched/cld2_api.h	2026-09-13 14:28:21.538318230 +0200
@@ -0,0 +1,18 @@
+#ifndef CLD2_API_H
+#define CLD2_API_H
+
+#define CLD2_API_EXPORT __attribute__((visibility("default")))
+#define CLD2_SUCCESS 0
+#define CLD2_ERROR 1
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+CLD2_API_EXPORT const char* cld2_detect_language(const char* buffer, int buffer_length);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif  // CLD2_API_H
diff -ruN -x '.*' cld2-org/CMakeLists.txt cld2-patched/CMakeLists.txt
--- cld2-org/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ cld2-patched/CMakeLists.txt	2026-09-13 14:50:01.668174510 +0200
@@ -0,0 +1,50 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(cld2_api LANGUAGES CXX VERSION 1.0.0)
+
+set(BUILD_SHARED_LIBS OFF)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+add_compile_options(-Wno-non-c-typedef-for-linkage -Wno-narrowing)
+
+set(cld2_source_dir "${PROJECT_SOURCE_DIR}/internal")
+set(cld2_include_dir "${PROJECT_SOURCE_DIR}/public")
+set(cld2_sources
+    ${cld2_source_dir}/cldutil.cc
+    ${cld2_source_dir}/cldutil_shared.cc
+    ${cld2_source_dir}/compact_lang_det.cc
+    ${cld2_source_dir}/compact_lang_det_hint_code.cc
+    ${cld2_source_dir}/compact_lang_det_impl.cc
+    ${cld2_source_dir}/debug.cc
+    ${cld2_source_dir}/fixunicodevalue.cc
+    ${cld2_source_dir}/generated_entities.cc
+    ${cld2_source_dir}/generated_language.cc
+    ${cld2_source_dir}/generated_ulscript.cc
+    ${cld2_source_dir}/getonescriptspan.cc
+    ${cld2_source_dir}/lang_script.cc
+    ${cld2_source_dir}/offsetmap.cc
+    ${cld2_source_dir}/scoreonescriptspan.cc
+    ${cld2_source_dir}/tote.cc
+    ${cld2_source_dir}/utf8statetable.cc
+    ${cld2_source_dir}/cld_generated_cjk_uni_prop_80.cc
+    ${cld2_source_dir}/cld2_generated_cjk_compatible.cc
+    ${cld2_source_dir}/cld_generated_cjk_delta_bi_4.cc
+    ${cld2_source_dir}/generated_distinct_bi_0.cc
+    ${cld2_source_dir}/cld2_generated_quadchrome_2.cc
+    ${cld2_source_dir}/cld2_generated_deltaoctachrome.cc
+    ${cld2_source_dir}/cld2_generated_distinctoctachrome.cc
+    ${cld2_source_dir}/cld_generated_score_quad_octa_2.cc
+)
+
+add_library(${CMAKE_PROJECT_NAME} STATIC cld2_api.cpp ${cld2_sources})
+
+target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE 
+    ${PROJECT_SOURCE_DIR}
+    ${cld2_source_dir}
+    ${cld2_include_dir})
+set_target_properties(${CMAKE_PROJECT_NAME} PROPERTIES PUBLIC_HEADER "cld2_api.h")
+
+install(TARGETS ${CMAKE_PROJECT_NAME}
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
+    PUBLIC_HEADER DESTINATION include)
diff -ruN -x '.*' cld2-org/public/compact_lang_det.h cld2-patched/public/compact_lang_det.h
--- cld2-org/public/compact_lang_det.h	2015-08-21 00:39:36.000000000 +0200
+++ cld2-patched/public/compact_lang_det.h	2026-09-13 14:54:24.626929119 +0200
@@ -62,7 +62,8 @@
 #ifndef I18N_ENCODINGS_CLD2_PUBLIC_COMPACT_LANG_DET_H_
 #define I18N_ENCODINGS_CLD2_PUBLIC_COMPACT_LANG_DET_H_
 
-#include <stdint.h>
+#include <cstdint>
+#include <cstdio>
 #include <vector>
 #include "../internal/integral_types.h"   // For uint8 etc.
 #include "../internal/lang_script.h"      // For Language
