diff -ruN -x '.*' espeak-org/cmake/deps.cmake espeak-patched-flatpak/cmake/deps.cmake
--- espeak-org/cmake/deps.cmake	2026-04-27 15:36:42.000000000 +0200
+++ espeak-patched-flatpak/cmake/deps.cmake	2026-07-11 17:43:20.867492629 +0200
@@ -17,19 +17,7 @@
 if (SONIC_LIB AND SONIC_INC)
   set(HAVE_LIBSONIC ON)
 else()
-  FetchContent_Declare(sonic-git
-    GIT_REPOSITORY https://github.com/waywardgeek/sonic.git
-    GIT_TAG fbf75c3d6d846bad3bb3d456cbc5d07d9fd8c104
-  )
-  FetchContent_MakeAvailable(sonic-git)
-  FetchContent_GetProperties(sonic-git)
-  add_library(sonic OBJECT ${sonic-git_SOURCE_DIR}/sonic.c)
-  # enable PIC in order to embed libsonic.a in a .so
-  set_property(TARGET sonic PROPERTY POSITION_INDEPENDENT_CODE ON)
-  target_include_directories(sonic PUBLIC ${sonic-git_SOURCE_DIR})
-  set(HAVE_LIBSONIC ON)
-  set(SONIC_LIB sonic)
-  set(SONIC_INC ${sonic-git_SOURCE_DIR})
+  set(HAVE_LIBSONIC OFF)
 endif()
 if (PCAUDIO_LIB AND PCAUDIO_INC)
   set(HAVE_LIBPCAUDIO ON)
diff -ruN -x '.*' espeak-org/src/libespeak-ng/CMakeLists.txt espeak-patched-flatpak/src/libespeak-ng/CMakeLists.txt
--- espeak-org/src/libespeak-ng/CMakeLists.txt	2026-04-27 15:36:42.000000000 +0200
+++ espeak-patched-flatpak/src/libespeak-ng/CMakeLists.txt	2026-07-11 17:08:26.797525718 +0200
@@ -69,7 +69,7 @@
   target_compile_definitions(espeak-ng INTERFACE "LIBESPEAK_NG_EXPORT=1")
 endif()
 
-target_compile_definitions(espeak-ng PRIVATE "PATH_ESPEAK_DATA=\"${CMAKE_INSTALL_PREFIX}/share/espeak-ng-data\"")
+target_compile_definitions(espeak-ng PRIVATE "PATH_ESPEAK_DATA=\"/app/share/espeak-ng-data\"")
 
 if (USE_ASYNC)
   target_sources(espeak-ng PRIVATE
