diff -ruN html2md-org/include/html2md.h html2md-patched/include/html2md.h
--- html2md-org/include/html2md.h	2025-02-13 09:38:03.000000000 +0100
+++ html2md-patched/include/html2md.h	2025-05-10 20:17:11.600032700 +0200
@@ -7,6 +7,7 @@
 #include <memory>
 #include <string>
 #include <unordered_map>
+#include <cstdint>
 
 /*!
  * \brief html2md namespace
diff -ruN html2md-org/src/html2md.cpp html2md-patched/src/html2md.cpp
--- html2md-org/src/html2md.cpp	2025-02-13 09:38:03.000000000 +0100
+++ html2md-patched/src/html2md.cpp	2025-05-10 20:15:44.589932259 +0200
@@ -186,8 +186,7 @@
   return this;
 }
 
-Converter *Converter::appendToMd(const char *str)
-{
+Converter *Converter::appendToMd(const char *str) {
   if (IsInIgnoredTag())
     return this;
 
@@ -607,11 +606,11 @@
 }
 
 void Converter::TagStrikethrought::OnHasLeftOpeningTag(Converter *c) {
-  c->appendToMd('~');
+  c->appendToMd("~~");
 }
 
 void Converter::TagStrikethrought::OnHasLeftClosingTag(Converter *c) {
-  c->appendToMd('~');
+  c->appendToMd("~~");
 }
 
 void Converter::TagBreak::OnHasLeftOpeningTag(Converter *c) {
