diff -ruN -x '.*' html2md-org/include/html2md.h html2md-patched/include/html2md.h
--- html2md-org/include/html2md.h	2025-05-14 09:55:21.000000000 +0200
+++ html2md-patched/include/html2md.h	2026-07-11 19:09:00.879484504 +0200
@@ -7,6 +7,7 @@
 #include <memory>
 #include <string>
 #include <unordered_map>
+#include <cstdint>
 
 /*!
  * \brief html2md namespace
diff -ruN -x '.*' html2md-org/src/html2md.cpp html2md-patched/src/html2md.cpp
--- html2md-org/src/html2md.cpp	2025-05-14 09:55:21.000000000 +0200
+++ html2md-patched/src/html2md.cpp	2026-07-11 19:12:39.605633296 +0200
@@ -671,11 +671,11 @@
 }
 
 void Converter::TagStrikethrought::OnHasLeftOpeningTag(Converter *c) {
-  c->appendToMd('~');
+  c->appendToMd("~~");
 }
 
 void Converter::TagStrikethrought::OnHasLeftClosingTag(Converter *c) {
-  c->appendToMd('~');
+  c->appendToMd("~~");
 }
 
 void Converter::TagBreak::OnHasLeftOpeningTag(Converter *c) {
