diff -ruN openfst-org/src/include/fst/fst.h openfst-patched/src/include/fst/fst.h
--- openfst-org/src/include/fst/fst.h	2021-02-07 09:38:45.000000000 +0100
+++ openfst-patched/src/include/fst/fst.h	2024-10-06 17:38:03.698500302 +0200
@@ -701,8 +701,8 @@
     properties_.store(impl.properties_.load(std::memory_order_relaxed),
                       std::memory_order_relaxed);
     type_ = impl.type_;
-    isymbols_ = impl.isymbols_ ? impl.isymbols_->Copy() : nullptr;
-    osymbols_ = impl.osymbols_ ? impl.osymbols_->Copy() : nullptr;
+    isymbols_ = fst::WrapUnique(impl.isymbols_ ? impl.isymbols_->Copy() : nullptr);
+    osymbols_ = fst::WrapUnique(impl.osymbols_ ? impl.osymbols_->Copy() : nullptr);
     return *this;
   }
 
