diff -ruN -x '.*' openfst-org/src/include/fst/bi-table.h openfst-patched/src/include/fst/bi-table.h
--- openfst-org/src/include/fst/bi-table.h	2024-08-23 16:17:27.000000000 +0200
+++ openfst-patched/src/include/fst/bi-table.h	2026-08-20 17:01:54.588245870 +0200
@@ -34,6 +34,7 @@
 #include <fst/windows_defs.inc>
 #include <unordered_map>
 #include <unordered_set>
+#include <cstdint>
 
 namespace fst {
 
