diff -ruN piperphonemize-org/CMakeLists.txt piperphonemize-patched/CMakeLists.txt
--- piperphonemize-org/CMakeLists.txt	2023-08-01 00:06:57.000000000 +0200
+++ piperphonemize-patched/CMakeLists.txt	2023-08-01 10:12:16.409075260 +0200
@@ -13,19 +13,19 @@
 string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -Wl,-rpath,'$ORIGIN'")
 string(APPEND CMAKE_C_FLAGS " -Wall -Wextra")
 
-# lib/Linux-x86_64
-# lib/Linux-aarch64
-set(ONNXRUNTIME_ROOTDIR ${CMAKE_CURRENT_LIST_DIR}/lib/${CMAKE_HOST_SYSTEM_NAME}-${CMAKE_HOST_SYSTEM_PROCESSOR}/onnxruntime)
-
 # ---- espeak-ng ---
 
 find_package(PkgConfig)
 pkg_check_modules(ESPEAK_NG REQUIRED espeak-ng<2)
 
+# ---- onnxruntime ---
+
+find_library(onnxruntime_path onnxruntime REQUIRED)
+
 # ---- Declare library ----
 
 add_library(
-    piper_phonemize SHARED
+    piper_phonemize
     src/phonemize.cpp
     src/phoneme_ids.cpp
     src/tashkeel.cpp
@@ -39,19 +39,18 @@
     piper_phonemize PUBLIC
     "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>"
     ${ESPEAK_NG_INCLUDE_DIRS}
-    ${ONNXRUNTIME_ROOTDIR}/include
 )
 
 target_link_directories(
     piper_phonemize PUBLIC
     ${ESPEAK_NG_LIBRARY_DIRS}
-    ${ONNXRUNTIME_ROOTDIR}/lib
+
 )
 
 target_link_libraries(
     piper_phonemize
     ${ESPEAK_NG_LIBRARIES}
-    onnxruntime
+    ${onnxruntime_path}
 )
 
 target_compile_options(
@@ -61,38 +60,10 @@
 
 target_compile_features(piper_phonemize PUBLIC cxx_std_17)
 
-# ---- Declare executable ----
-
-add_executable(piper_phonemize_exe src/main.cpp)
-set_property(TARGET piper_phonemize_exe PROPERTY OUTPUT_NAME piper_phonemize)
-
-target_compile_features(piper_phonemize_exe PUBLIC cxx_std_17)
-
-target_include_directories(
-    piper_phonemize_exe PUBLIC
-    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>"
-    ${ESPEAK_NG_INCLUDE_DIRS}
-)
-
-target_link_directories(
-    piper_phonemize_exe PUBLIC
-    ${ESPEAK_NG_LIBRARY_DIRS}
-)
-
-target_link_libraries(piper_phonemize_exe PUBLIC
-                      piper_phonemize
-                      ${ESPEAK_NG_LIBRARIES})
-
-target_compile_options(
-    piper_phonemize_exe PUBLIC
-    ${ESPEAK_NG_CFLAGS_OTHER}
-)
-
-# ---- Declare install targets ----
-
 install(
     TARGETS piper_phonemize
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib)
 
 install(
     DIRECTORY ${CMAKE_SOURCE_DIR}/src/
@@ -100,7 +71,3 @@
     FILES_MATCHING
     PATTERN "*.h"
     PATTERN "*.hpp")
-
-install(
-    TARGETS piper_phonemize_exe
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_BINDIR})
