diff -ruN -x '.*' piperphonemize-org/CMakeLists.txt piperphonemize-patched/CMakeLists.txt
--- piperphonemize-org/CMakeLists.txt	2025-07-11 00:00:35.000000000 +0200
+++ piperphonemize-patched/CMakeLists.txt	2026-06-04 20:38:44.470334200 +0200
@@ -22,7 +22,7 @@
 endif()
 
 add_library(
-    piper_phonemize SHARED
+    piper_phonemize
     src/phonemize.cpp
     src/phoneme_ids.cpp
     src/tashkeel.cpp
@@ -144,67 +144,14 @@
 
 target_compile_features(piper_phonemize PUBLIC cxx_std_17)
 
-# ---- Declare executable ----
-
-add_executable(piper_phonemize_exe src/main.cpp src/phoneme_ids.cpp)
-
-if(NOT WIN32)
-    set_property(TARGET piper_phonemize_exe PROPERTY OUTPUT_NAME piper_phonemize)
-endif()
-
-target_compile_features(piper_phonemize_exe PUBLIC cxx_std_17)
-
-target_include_directories(
-    piper_phonemize_exe PUBLIC
-    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>"
-    ${ESPEAK_NG_DIR}/include
-)
-
-target_link_directories(
-    piper_phonemize_exe PUBLIC
-    ${ESPEAK_NG_DIR}/lib
-)
-
-target_link_libraries(piper_phonemize_exe PUBLIC
-    piper_phonemize
-    espeak-ng
-)
-
-# ---- Declare test ----
-
-include(CTest)
-enable_testing()
-add_executable(test_piper_phonemize src/test.cpp src/phoneme_ids.cpp)
-add_test(
-    NAME test_piper_phonemize
-    COMMAND test_piper_phonemize "${ESPEAK_NG_DIR}/share/espeak-ng-data" "${CMAKE_SOURCE_DIR}/etc/libtashkeel_model.ort"
-)
-
-target_compile_features(test_piper_phonemize PUBLIC cxx_std_17)
-
-target_include_directories(
-    test_piper_phonemize PUBLIC
-    "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>"
-    ${ESPEAK_NG_DIR}/include
-)
-
-target_link_directories(
-    test_piper_phonemize PUBLIC
-    ${ESPEAK_NG_DIR}/lib
-)
-
-target_link_libraries(test_piper_phonemize PUBLIC
-    piper_phonemize
-    espeak-ng
-)
-
 # ---- Declare install targets ----
 
 include(GNUInstallDirs)
 
 install(
     TARGETS piper_phonemize
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib)
 
 install(
     DIRECTORY ${CMAKE_SOURCE_DIR}/src/
@@ -212,24 +159,3 @@
     FILES_MATCHING
     PATTERN "*.h"
     PATTERN "*.hpp")
-
-install(
-    TARGETS piper_phonemize_exe
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_BINDIR})
-
-install(
-    FILES ${CMAKE_SOURCE_DIR}/etc/libtashkeel_model.ort
-    TYPE DATA)
-
-# Dependencies
-install(
-    DIRECTORY ${ESPEAK_NG_DIR}/
-    DESTINATION ${CMAKE_INSTALL_PREFIX})
-
-install(
-    DIRECTORY ${ONNXRUNTIME_DIR}/include/
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-
-install(
-    DIRECTORY ${ONNXRUNTIME_DIR}/lib/
-    DESTINATION ${CMAKE_INSTALL_LIBDIR})
diff -ruN -x '.*' piperphonemize-org/src/phoneme_ids.hpp piperphonemize-patched/src/phoneme_ids.hpp
--- piperphonemize-org/src/phoneme_ids.hpp	2025-07-11 00:00:35.000000000 +0200
+++ piperphonemize-patched/src/phoneme_ids.hpp	2026-06-04 21:14:34.649807002 +0200
@@ -202,6 +202,10 @@
     // Czech
     {U'\u031d', {157}},
     {U'\u030a', {158}},
+    
+    // Latvian
+    {U'\u0361', {159}},
+    {U'\u0060', {160}},
 };
 
 // language -> phoneme -> [id, ...]
