diff -ruN org/TTS-0.18.2.dist-info/METADATA tmp/TTS-0.18.2.dist-info/METADATA
--- org/TTS-0.18.2.dist-info/METADATA	2023-10-06 22:23:36.000000000 +0200
+++ tmp/TTS-0.18.2.dist-info/METADATA	2023-10-20 19:17:30.243669302 +0200
@@ -29,7 +29,6 @@
 Requires-Python: >=3.9.0, <3.12
 Description-Content-Type: text/markdown
 License-File: LICENSE.txt
-Requires-Dist: cython ==0.29.30
 Requires-Dist: scipy >=1.11.2
 Requires-Dist: torch >=1.7
 Requires-Dist: torchaudio
