diff -ruN tmp/TTS-0.16.3.dist-info/METADATA tmp-patched/TTS-0.16.3.dist-info/METADATA
--- tmp/TTS-0.16.3.dist-info/METADATA	2023-08-13 10:26:28.000000000 +0200
+++ tmp-patched/TTS-0.16.3.dist-info/METADATA	2023-08-15 14:30:15.075722860 +0200
@@ -29,7 +29,7 @@
 Requires-Python: >=3.9.0, <3.12
 Description-Content-Type: text/markdown
 License-File: LICENSE.txt
-Requires-Dist: cython ==0.29.30
+Requires-Dist: cython
 Requires-Dist: scipy >=1.4.0
 Requires-Dist: torch >=1.7
 Requires-Dist: torchaudio
