diff -ruN -x '.*' whispercpp-org/ggml/include/ggml-backend.h whispercpp-patched/ggml/include/ggml-backend.h
--- whispercpp-org/ggml/include/ggml-backend.h	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/include/ggml-backend.h	2026-08-01 15:24:38.818956070 +0200
@@ -251,6 +251,8 @@
 
     // Load a backend from a dynamic library and register it
     GGML_API ggml_backend_reg_t ggml_backend_load(const char * path);
+    // Load a backend from a dynamic library and register it
+    GGML_API ggml_backend_reg_t ggml_backend_load_best_ex(const char * name);
     // Unload a backend if loaded dynamically and unregister it
     GGML_API void               ggml_backend_unload(ggml_backend_reg_t reg);
     // Load all known backends from dynamic libraries
diff -ruN -x '.*' whispercpp-org/ggml/src/CMakeLists.txt whispercpp-patched/ggml/src/CMakeLists.txt
--- whispercpp-org/ggml/src/CMakeLists.txt	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/src/CMakeLists.txt	2026-08-16 17:56:29.790179972 +0200
@@ -405,14 +405,41 @@
             # Many of these features are optional so we build versions with popular
             # combinations and name the backends based on the version they were
             # first released with
-            ggml_add_cpu_backend_variant(armv8.0_1)
-            ggml_add_cpu_backend_variant(armv8.2_1    DOTPROD)
-            ggml_add_cpu_backend_variant(armv8.2_2    DOTPROD FP16_VECTOR_ARITHMETIC)
-            ggml_add_cpu_backend_variant(armv8.2_3    DOTPROD FP16_VECTOR_ARITHMETIC SVE)
-            ggml_add_cpu_backend_variant(armv8.6_1    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8)
-            ggml_add_cpu_backend_variant(armv8.6_2    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2)
-            ggml_add_cpu_backend_variant(armv9.2_1    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SME)
-            ggml_add_cpu_backend_variant(armv9.2_2    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2 SME)
+            if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+                if (CMAKE_SYSTEM_PROCESSOR MATCHES "armv7")
+                    ggml_add_cpu_backend_variant(armv7.0_1)
+                    ggml_add_cpu_backend_variant(armv7.0_2 NEON)
+                else()
+                    ggml_add_cpu_backend_variant(armv8.0_1)
+                    if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 9)
+                        ggml_add_cpu_backend_variant(armv8.2_1 DOTPROD)
+                        ggml_add_cpu_backend_variant(armv8.2_2 DOTPROD FP16_VECTOR_ARITHMETIC)
+                    endif()
+                    if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
+                        ggml_add_cpu_backend_variant(armv8.2_3 DOTPROD FP16_VECTOR_ARITHMETIC SVE)
+                    endif()
+                    if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 11)
+                        ggml_add_cpu_backend_variant(armv8.6_1 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8)
+                        ggml_add_cpu_backend_variant(armv8.6_2 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2)
+                    endif()
+                    if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 14)
+                        ggml_add_cpu_backend_variant(armv9.2_1 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SME)
+                        ggml_add_cpu_backend_variant(armv9.2_2 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2 SME)
+                    elseif(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 13)
+                        ggml_add_cpu_backend_variant(armv9.2_1 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8)
+                        ggml_add_cpu_backend_variant(armv9.2_2 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2)
+                    endif()
+                endif()
+            else()
+                ggml_add_cpu_backend_variant(armv8.0_1)
+                ggml_add_cpu_backend_variant(armv8.2_1    DOTPROD)
+                ggml_add_cpu_backend_variant(armv8.2_2    DOTPROD FP16_VECTOR_ARITHMETIC)
+                ggml_add_cpu_backend_variant(armv8.2_3    DOTPROD FP16_VECTOR_ARITHMETIC SVE)
+                ggml_add_cpu_backend_variant(armv8.6_1    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8)
+                ggml_add_cpu_backend_variant(armv8.6_2    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2)
+                ggml_add_cpu_backend_variant(armv9.2_1    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SME)
+                ggml_add_cpu_backend_variant(armv9.2_2    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2 SME)
+            endif()
         elseif (CMAKE_SYSTEM_NAME MATCHES "Android")
             # Android-specific backends with SoC-compatible feature sets
             ggml_add_cpu_backend_variant(android_armv8.0_1)
diff -ruN -x '.*' whispercpp-org/ggml/src/ggml-backend-reg.cpp whispercpp-patched/ggml/src/ggml-backend-reg.cpp
--- whispercpp-org/ggml/src/ggml-backend-reg.cpp	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/src/ggml-backend-reg.cpp	2026-08-16 15:54:52.678222989 +0200
@@ -478,9 +478,10 @@
 
     std::vector<fs::path> search_paths;
     if (user_search_path == nullptr) {
-#ifdef GGML_BACKEND_DIR
-        search_paths.push_back(fs::u8path(GGML_BACKEND_DIR));
-#endif
+        auto* env_path = getenv("GGML_BACKEND_DIR");
+        if (env_path != nullptr && strlen(env_path) > 0) {
+            search_paths.emplace_back(env_path);
+        }
         // default search paths: executable directory, current directory
         search_paths.push_back(get_executable_path());
         search_paths.push_back(fs::current_path());
@@ -556,6 +557,10 @@
     ggml_backend_load_all_from_path(nullptr);
 }
 
+ggml_backend_reg_t ggml_backend_load_best_ex(const char* name) {
+    return ggml_backend_load_best(name, false, nullptr);
+}
+
 void ggml_backend_load_all_from_path(const char * dir_path) {
 #ifdef NDEBUG
     bool silent = true;
diff -ruN -x '.*' whispercpp-org/ggml/src/ggml-cpu/CMakeLists.txt whispercpp-patched/ggml/src/ggml-cpu/CMakeLists.txt
--- whispercpp-org/ggml/src/ggml-cpu/CMakeLists.txt	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/src/ggml-cpu/CMakeLists.txt	2026-08-16 17:50:01.683295363 +0200
@@ -169,45 +169,57 @@
                 if (GGML_CPU_ARM_ARCH)
                     list(APPEND ARCH_FLAGS -march=${GGML_CPU_ARM_ARCH})
                 elseif(GGML_CPU_ALL_VARIANTS)
-                    # Begin with the lowest baseline
-                    set(ARM_MCPU "armv8-a")
-                    set(ARCH_TAGS "")
-                    set(ARCH_DEFINITIONS "")
+                    if (CMAKE_SYSTEM_PROCESSOR MATCHES "armv7")
+                        # Begin with the lowest baseline
+                        set(ARM_MCPU "armv7-a")
+                        set(ARCH_TAGS "")
+                        set(ARCH_DEFINITIONS "")
 
-                    # When a feature is selected, bump the MCPU to the first
-                    # version that supported it
-                    if (GGML_INTERNAL_DOTPROD)
-                        set(ARM_MCPU "armv8.2-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+dotprod")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_DOTPROD)
-                    endif()
-                    if (GGML_INTERNAL_FP16_VECTOR_ARITHMETIC)
-                        set(ARM_MCPU "armv8.2-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+fp16")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_FP16_VECTOR_ARITHMETIC)
-                    endif()
-                    if (GGML_INTERNAL_SVE)
-                        set(ARM_MCPU "armv8.2-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+sve")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_SVE)
-                    endif()
-                    if (GGML_INTERNAL_MATMUL_INT8)
-                        set(ARM_MCPU "armv8.6-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+i8mm")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_MATMUL_INT8)
-                    endif()
-                    if (GGML_INTERNAL_SVE2)
-                        set(ARM_MCPU "armv8.6-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+sve2")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_SVE2)
-                    endif()
-                    if (GGML_INTERNAL_NOSVE)
-                        set(ARCH_TAGS "${ARCH_TAGS}+nosve")
-                    endif()
-                    if (GGML_INTERNAL_SME)
-                        set(ARM_MCPU "armv9.2-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+sme")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_SME)
+                        if (GGML_INTERNAL_NEON)
+                            set(ARM_MCPU "armv7-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+fp+simd")
+                        endif()
+                    else()
+                        # Begin with the lowest baseline
+                        set(ARM_MCPU "armv8-a")
+                        set(ARCH_TAGS "")
+                        set(ARCH_DEFINITIONS "")
+
+                        # When a feature is selected, bump the MCPU to the first
+                        # version that supported it
+                        if (GGML_INTERNAL_DOTPROD)
+                            set(ARM_MCPU "armv8.2-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+dotprod")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_DOTPROD)
+                        endif()
+                        if (GGML_INTERNAL_FP16_VECTOR_ARITHMETIC)
+                            set(ARM_MCPU "armv8.2-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+fp16")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_FP16_VECTOR_ARITHMETIC)
+                        endif()
+                        if (GGML_INTERNAL_SVE)
+                            set(ARM_MCPU "armv8.2-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+sve")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_SVE)
+                        endif()
+                        if (GGML_INTERNAL_MATMUL_INT8)
+                            set(ARM_MCPU "armv8.6-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+i8mm")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_MATMUL_INT8)
+                        endif()
+                        if (GGML_INTERNAL_SVE2)
+                            set(ARM_MCPU "armv8.6-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+sve2")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_SVE2)
+                        endif()
+                        if (GGML_INTERNAL_NOSVE)
+                            set(ARCH_TAGS "${ARCH_TAGS}+nosve")
+                        endif()
+                        if (GGML_INTERNAL_SME)
+                            set(ARM_MCPU "armv9.2-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+sme")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_SME)
+                        endif()
                     endif()
                     list(APPEND ARCH_FLAGS "-march=${ARM_MCPU}${ARCH_TAGS}")
                     ggml_add_cpu_backend_features(${GGML_CPU_NAME} arm ${ARCH_DEFINITIONS})
