diff -ruN -x '.*' whispercpp-org/ggml/include/ggml-backend.h whispercpp-patched/ggml/include/ggml-backend.h
--- whispercpp-org/ggml/include/ggml-backend.h	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/include/ggml-backend.h	2026-08-01 15:24:38.818956070 +0200
@@ -251,6 +251,8 @@
 
     // Load a backend from a dynamic library and register it
     GGML_API ggml_backend_reg_t ggml_backend_load(const char * path);
+    // Load a backend from a dynamic library and register it
+    GGML_API ggml_backend_reg_t ggml_backend_load_best_ex(const char * name);
     // Unload a backend if loaded dynamically and unregister it
     GGML_API void               ggml_backend_unload(ggml_backend_reg_t reg);
     // Load all known backends from dynamic libraries
diff -ruN -x '.*' whispercpp-org/ggml/src/CMakeLists.txt whispercpp-patched/ggml/src/CMakeLists.txt
--- whispercpp-org/ggml/src/CMakeLists.txt	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/src/CMakeLists.txt	2026-08-20 11:06:07.394293636 +0200
@@ -405,14 +405,42 @@
             # Many of these features are optional so we build versions with popular
             # combinations and name the backends based on the version they were
             # first released with
-            ggml_add_cpu_backend_variant(armv8.0_1)
-            ggml_add_cpu_backend_variant(armv8.2_1    DOTPROD)
-            ggml_add_cpu_backend_variant(armv8.2_2    DOTPROD FP16_VECTOR_ARITHMETIC)
-            ggml_add_cpu_backend_variant(armv8.2_3    DOTPROD FP16_VECTOR_ARITHMETIC SVE)
-            ggml_add_cpu_backend_variant(armv8.6_1    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8)
-            ggml_add_cpu_backend_variant(armv8.6_2    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2)
-            ggml_add_cpu_backend_variant(armv9.2_1    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SME)
-            ggml_add_cpu_backend_variant(armv9.2_2    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2 SME)
+            if(CMAKE_C_COMPILER_ID STREQUAL "GNU")
+                if (CMAKE_SYSTEM_PROCESSOR MATCHES "armv7")
+                    ggml_add_cpu_backend_variant(armv7.0_1)
+                    ggml_add_cpu_backend_variant(armv7.0_2 ARM_VFPV4 ARM_NEON)
+                    ggml_add_cpu_backend_variant(armv8.0_1 ARM_VFPV4 ARM_NEON ARM8_NEON)
+                else()
+                    ggml_add_cpu_backend_variant(armv8.0_1)
+                    if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 9)
+                        ggml_add_cpu_backend_variant(armv8.2_1 DOTPROD)
+                        ggml_add_cpu_backend_variant(armv8.2_2 DOTPROD FP16_VECTOR_ARITHMETIC)
+                    endif()
+                    if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
+                        ggml_add_cpu_backend_variant(armv8.2_3 DOTPROD FP16_VECTOR_ARITHMETIC SVE)
+                    endif()
+                    if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 11)
+                        ggml_add_cpu_backend_variant(armv8.6_1 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8)
+                        ggml_add_cpu_backend_variant(armv8.6_2 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2)
+                    endif()
+                    if(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 14)
+                        ggml_add_cpu_backend_variant(armv9.2_1 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SME)
+                        ggml_add_cpu_backend_variant(armv9.2_2 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2 SME)
+                    elseif(CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 13)
+                        ggml_add_cpu_backend_variant(armv9.2_1 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8)
+                        ggml_add_cpu_backend_variant(armv9.2_2 DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2)
+                    endif()
+                endif()
+            else()
+                ggml_add_cpu_backend_variant(armv8.0_1)
+                ggml_add_cpu_backend_variant(armv8.2_1    DOTPROD)
+                ggml_add_cpu_backend_variant(armv8.2_2    DOTPROD FP16_VECTOR_ARITHMETIC)
+                ggml_add_cpu_backend_variant(armv8.2_3    DOTPROD FP16_VECTOR_ARITHMETIC SVE)
+                ggml_add_cpu_backend_variant(armv8.6_1    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8)
+                ggml_add_cpu_backend_variant(armv8.6_2    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2)
+                ggml_add_cpu_backend_variant(armv9.2_1    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SME)
+                ggml_add_cpu_backend_variant(armv9.2_2    DOTPROD FP16_VECTOR_ARITHMETIC SVE MATMUL_INT8 SVE2 SME)
+            endif()
         elseif (CMAKE_SYSTEM_NAME MATCHES "Android")
             # Android-specific backends with SoC-compatible feature sets
             ggml_add_cpu_backend_variant(android_armv8.0_1)
diff -ruN -x '.*' whispercpp-org/ggml/src/ggml-backend-reg.cpp whispercpp-patched/ggml/src/ggml-backend-reg.cpp
--- whispercpp-org/ggml/src/ggml-backend-reg.cpp	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/src/ggml-backend-reg.cpp	2026-08-16 15:54:52.678222989 +0200
@@ -478,9 +478,10 @@
 
     std::vector<fs::path> search_paths;
     if (user_search_path == nullptr) {
-#ifdef GGML_BACKEND_DIR
-        search_paths.push_back(fs::u8path(GGML_BACKEND_DIR));
-#endif
+        auto* env_path = getenv("GGML_BACKEND_DIR");
+        if (env_path != nullptr && strlen(env_path) > 0) {
+            search_paths.emplace_back(env_path);
+        }
         // default search paths: executable directory, current directory
         search_paths.push_back(get_executable_path());
         search_paths.push_back(fs::current_path());
@@ -556,6 +557,10 @@
     ggml_backend_load_all_from_path(nullptr);
 }
 
+ggml_backend_reg_t ggml_backend_load_best_ex(const char* name) {
+    return ggml_backend_load_best(name, false, nullptr);
+}
+
 void ggml_backend_load_all_from_path(const char * dir_path) {
 #ifdef NDEBUG
     bool silent = true;
diff -ruN -x '.*' whispercpp-org/ggml/src/ggml-blas/ggml-blas.cpp whispercpp-patched/ggml/src/ggml-blas/ggml-blas.cpp
--- whispercpp-org/ggml/src/ggml-blas/ggml-blas.cpp	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/src/ggml-blas/ggml-blas.cpp	2026-08-16 19:57:04.565343972 +0200
@@ -14,6 +14,8 @@
 #   include <blis.h>
 #elif defined(GGML_BLAS_USE_NVPL)
 #   include <nvpl_blas.h>
+#elif defined(GGML_BLAS_USE_OPENBLAS)
+#   include <openblas/cblas.h>
 #else
 #   include <cblas.h>
 #endif
diff -ruN -x '.*' whispercpp-org/ggml/src/ggml-cpu/arch/arm/cpu-feats.cpp whispercpp-patched/ggml/src/ggml-cpu/arch/arm/cpu-feats.cpp
--- whispercpp-org/ggml/src/ggml-cpu/arch/arm/cpu-feats.cpp	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/src/ggml-cpu/arch/arm/cpu-feats.cpp	2026-08-20 11:09:13.432361517 +0200
@@ -22,12 +22,12 @@
 
 struct aarch64_features {
     // has_neon not needed, aarch64 has NEON guaranteed
-    bool has_dotprod     = false;
-    bool has_fp16_va     = false;
-    bool has_sve         = false;
-    bool has_sve2        = false;
-    bool has_i8mm        = false;
-    bool has_sme         = false;
+    bool has_dotprod = false;
+    bool has_fp16_va = false;
+    bool has_sve = false;
+    bool has_sve2 = false;
+    bool has_i8mm = false;
+    bool has_sme = false;
 
     aarch64_features() {
 #if defined(__linux__)
@@ -36,23 +36,26 @@
 
         has_dotprod = !!(hwcap & HWCAP_ASIMDDP);
         has_fp16_va = !!(hwcap & HWCAP_FPHP);
-        has_sve     = !!(hwcap & HWCAP_SVE);
-        has_sve2    = !!(hwcap2 & HWCAP2_SVE2);
-        has_i8mm    = !!(hwcap2 & HWCAP2_I8MM);
-        has_sme     = !!(hwcap2 & HWCAP2_SME);
+        has_sve = !!(hwcap & HWCAP_SVE);
+        has_sve2 = !!(hwcap2 & HWCAP2_SVE2);
+        has_i8mm = !!(hwcap2 & HWCAP2_I8MM);
+        has_sme = !!(hwcap2 & HWCAP2_SME);
 #elif defined(__APPLE__)
         int oldp = 0;
         size_t size = sizeof(oldp);
 
-        if (sysctlbyname("hw.optional.arm.FEAT_DotProd", &oldp, &size, NULL, 0) == 0) {
+        if (sysctlbyname("hw.optional.arm.FEAT_DotProd", &oldp, &size, NULL,
+                         0) == 0) {
             has_dotprod = static_cast<bool>(oldp);
         }
 
-        if (sysctlbyname("hw.optional.arm.FEAT_I8MM", &oldp, &size, NULL, 0) == 0) {
+        if (sysctlbyname("hw.optional.arm.FEAT_I8MM", &oldp, &size, NULL, 0) ==
+            0) {
             has_i8mm = static_cast<bool>(oldp);
         }
 
-        if (sysctlbyname("hw.optional.arm.FEAT_SME", &oldp, &size, NULL, 0) == 0) {
+        if (sysctlbyname("hw.optional.arm.FEAT_SME", &oldp, &size, NULL, 0) ==
+            0) {
             has_sme = static_cast<bool>(oldp);
         }
 
@@ -66,28 +69,40 @@
     aarch64_features af;
 
 #ifdef GGML_USE_DOTPROD
-    if (!af.has_dotprod) { return 0; }
-    score += 1<<1;
+    if (!af.has_dotprod) {
+        return 0;
+    }
+    score += 1 << 1;
 #endif
 #ifdef GGML_USE_FP16_VECTOR_ARITHMETIC
-    if (!af.has_fp16_va) { return 0; }
-    score += 1<<2;
+    if (!af.has_fp16_va) {
+        return 0;
+    }
+    score += 1 << 2;
 #endif
 #ifdef GGML_USE_SVE
-    if (!af.has_sve) { return 0; }
-    score += 1<<3;
+    if (!af.has_sve) {
+        return 0;
+    }
+    score += 1 << 3;
 #endif
 #ifdef GGML_USE_MATMUL_INT8
-    if (!af.has_i8mm) { return 0; }
-    score += 1<<4;
+    if (!af.has_i8mm) {
+        return 0;
+    }
+    score += 1 << 4;
 #endif
 #ifdef GGML_USE_SVE2
-    if (!af.has_sve2) { return 0; }
-    score += 1<<5;
+    if (!af.has_sve2) {
+        return 0;
+    }
+    score += 1 << 5;
 #endif
 #ifdef GGML_USE_SME
-    if (!af.has_sme) { return 0; }
-    score += 1<<6;
+    if (!af.has_sme) {
+        return 0;
+    }
+    score += 1 << 6;
 #endif
 
     return score;
@@ -95,4 +110,127 @@
 
 GGML_BACKEND_DL_SCORE_IMPL(ggml_backend_cpu_aarch64_score)
 
-# endif // defined(__aarch64__)
+#endif  // defined(__aarch64__)
+
+#if defined(__arm__)
+#include <fstream>
+#include <regex>
+#include <string>
+
+#include <sys/auxv.h>
+
+#if !defined(HWCAP_NEON)
+#define HWCAP_NEON (1 << 12)
+#endif
+
+#if !defined(HWCAP_FPHP)
+#define HWCAP_FPHP (1 << 22)
+#endif
+
+#if !defined(HWCAP_ASIMDDP)
+#define HWCAP_ASIMDDP (1 << 24)
+#endif
+
+#if !defined(HWCAP_I8MM)
+#define HWCAP_I8MM (1 << 27)
+#endif
+
+static unsigned int get_cpu_arch_from_cpuinfo() {
+    std::ifstream cpuinfo_file{"/proc/cpuinfo"};
+    if (!cpuinfo_file) {
+        throw std::runtime_error{"can't open /proc/cpuinfo"};
+    }
+
+    static const std::regex arm_arch_rx{"CPU\\sarchitecture\\s*:\\s+(.*)"};
+    for (std::string line; std::getline(cpuinfo_file, line);) {
+        if (std::smatch pieces_match;
+            std::regex_match(line, pieces_match, arm_arch_rx)) {
+            if (pieces_match[1].str().find('7') != std::string::npos)
+                return 7;
+            else if (pieces_match[1].str().find('8') != std::string::npos)
+                return 8;
+            else if (pieces_match[1].str().find('9') != std::string::npos)
+                return 9;
+        }
+    }
+
+    throw std::runtime_error{"invalid cpu arch"};
+}
+
+struct arm_features {
+    bool has_neon = false;
+    bool has_vfpv3 = false;
+    bool has_vfpv4 = false;
+    bool has_fp16_va = false;
+    bool has_dotprod = false;
+    bool has_i8mm = false;
+    unsigned int arch = 7; // default
+
+    arm_features() {
+      uint32_t hwcap = getauxval(AT_HWCAP);
+
+      has_neon = !!(hwcap & HWCAP_ARM_NEON);
+      has_vfpv3 = !!(hwcap & HWCAP_ARM_VFPv3);
+      has_vfpv4 = !!(hwcap & HWCAP_ARM_VFPv4);
+      has_fp16_va = !!(hwcap & HWCAP_FPHP);
+      has_dotprod = !!(hwcap & HWCAP_ASIMDDP);
+      has_i8mm = !!(hwcap & HWCAP_I8MM);
+
+      try {
+        arch = get_cpu_arch_from_cpuinfo();
+      } catch (...) {
+      }
+    }
+};
+
+static int ggml_backend_cpu_arm_score() {
+    int score = 1;
+    arm_features af;
+#ifdef GGML_USE_ARM_VFPV3
+    if (!af.has_vfpv3) {
+        return 0;
+    }
+    score += 1 << 1;
+#endif
+#ifdef GGML_USE_ARM_VFPV4
+    if (!af.has_vfpv4) {
+        return 0;
+    }
+    score += 1 << 2;
+#endif
+#ifdef GGML_USE_ARM_NEON
+    if (!af.has_neon) {
+        return 0;
+    }
+    score += 1 << 3;
+#endif
+#ifdef GGML_USE_ARM8_NEON
+    if (!af.has_neon || af.arch < 8) {
+        return 0;
+    }
+    score += 1 << 4;
+#endif
+#ifdef GGML_USE_DOTPROD
+    if (!af.has_dotprod) {
+      return 0;
+    }
+    score += 1 << 5;
+#endif
+#ifdef GGML_USE_FP16_VECTOR_ARITHMETIC
+    if (!af.has_fp16_va) {
+      return 0;
+    }
+    score += 1 << 6;
+#endif
+#ifdef GGML_USE_MATMUL_INT8
+    if (!af.has_i8mm) {
+      return 0;
+    }
+    score += 1 << 7;
+#endif
+    return score;
+}
+
+GGML_BACKEND_DL_SCORE_IMPL(ggml_backend_cpu_arm_score)
+
+#endif  // defined(__arm__)
diff -ruN -x '.*' whispercpp-org/ggml/src/ggml-cpu/CMakeLists.txt whispercpp-patched/ggml/src/ggml-cpu/CMakeLists.txt
--- whispercpp-org/ggml/src/ggml-cpu/CMakeLists.txt	2026-06-19 06:12:37.000000000 +0200
+++ whispercpp-patched/ggml/src/ggml-cpu/CMakeLists.txt	2026-08-20 11:04:00.045261769 +0200
@@ -169,45 +169,68 @@
                 if (GGML_CPU_ARM_ARCH)
                     list(APPEND ARCH_FLAGS -march=${GGML_CPU_ARM_ARCH})
                 elseif(GGML_CPU_ALL_VARIANTS)
-                    # Begin with the lowest baseline
-                    set(ARM_MCPU "armv8-a")
-                    set(ARCH_TAGS "")
-                    set(ARCH_DEFINITIONS "")
+                    if (CMAKE_SYSTEM_PROCESSOR MATCHES "armv7")
+                        # Begin with the lowest baseline
+                        set(ARM_MCPU "armv7-a")
+                        set(ARCH_TAGS "")
+                        set(ARCH_DEFINITIONS "")
 
-                    # When a feature is selected, bump the MCPU to the first
-                    # version that supported it
-                    if (GGML_INTERNAL_DOTPROD)
-                        set(ARM_MCPU "armv8.2-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+dotprod")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_DOTPROD)
-                    endif()
-                    if (GGML_INTERNAL_FP16_VECTOR_ARITHMETIC)
-                        set(ARM_MCPU "armv8.2-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+fp16")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_FP16_VECTOR_ARITHMETIC)
-                    endif()
-                    if (GGML_INTERNAL_SVE)
-                        set(ARM_MCPU "armv8.2-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+sve")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_SVE)
-                    endif()
-                    if (GGML_INTERNAL_MATMUL_INT8)
-                        set(ARM_MCPU "armv8.6-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+i8mm")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_MATMUL_INT8)
-                    endif()
-                    if (GGML_INTERNAL_SVE2)
-                        set(ARM_MCPU "armv8.6-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+sve2")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_SVE2)
-                    endif()
-                    if (GGML_INTERNAL_NOSVE)
-                        set(ARCH_TAGS "${ARCH_TAGS}+nosve")
-                    endif()
-                    if (GGML_INTERNAL_SME)
-                        set(ARM_MCPU "armv9.2-a")
-                        set(ARCH_TAGS "${ARCH_TAGS}+sme")
-                        list(APPEND ARCH_DEFINITIONS GGML_USE_SME)
+                        if (GGML_INTERNAL_ARM_VFPV4)
+                            set(ARM_MCPU "armv7-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+vfpv4")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_ARM_VFPV4)
+                        endif()
+                        if (GGML_INTERNAL_ARM_NEON)
+                            set(ARM_MCPU "armv7-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+simd")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_ARM_NEON)
+                        endif()
+                        if (GGML_INTERNAL_ARM8_NEON)
+                            set(ARM_MCPU "armv8-a")
+                            set(ARCH_TAGS "+simd")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_ARM8_NEON)
+                        endif()
+                    else()
+                        # Begin with the lowest baseline
+                        set(ARM_MCPU "armv8-a")
+                        set(ARCH_TAGS "")
+                        set(ARCH_DEFINITIONS "")
+
+                        # When a feature is selected, bump the MCPU to the first
+                        # version that supported it
+                        if (GGML_INTERNAL_DOTPROD)
+                            set(ARM_MCPU "armv8.2-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+dotprod")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_DOTPROD)
+                        endif()
+                        if (GGML_INTERNAL_FP16_VECTOR_ARITHMETIC)
+                            set(ARM_MCPU "armv8.2-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+fp16")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_FP16_VECTOR_ARITHMETIC)
+                        endif()
+                        if (GGML_INTERNAL_SVE)
+                            set(ARM_MCPU "armv8.2-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+sve")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_SVE)
+                        endif()
+                        if (GGML_INTERNAL_MATMUL_INT8)
+                            set(ARM_MCPU "armv8.6-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+i8mm")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_MATMUL_INT8)
+                        endif()
+                        if (GGML_INTERNAL_SVE2)
+                            set(ARM_MCPU "armv8.6-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+sve2")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_SVE2)
+                        endif()
+                        if (GGML_INTERNAL_NOSVE)
+                            set(ARCH_TAGS "${ARCH_TAGS}+nosve")
+                        endif()
+                        if (GGML_INTERNAL_SME)
+                            set(ARM_MCPU "armv9.2-a")
+                            set(ARCH_TAGS "${ARCH_TAGS}+sme")
+                            list(APPEND ARCH_DEFINITIONS GGML_USE_SME)
+                        endif()
                     endif()
                     list(APPEND ARCH_FLAGS "-march=${ARM_MCPU}${ARCH_TAGS}")
                     ggml_add_cpu_backend_features(${GGML_CPU_NAME} arm ${ARCH_DEFINITIONS})
