diff -ruN whispercpp-1.7.1-org/cmake/DefaultTargetOptions.cmake whispercpp-1.7.1-patched/cmake/DefaultTargetOptions.cmake
--- whispercpp-1.7.1-org/cmake/DefaultTargetOptions.cmake	2024-10-07 12:06:48.000000000 +0200
+++ whispercpp-1.7.1-patched/cmake/DefaultTargetOptions.cmake	2024-10-26 16:07:23.460491574 +0200
@@ -9,9 +9,13 @@
         cxx_std_11
     )
 
+if (NOT DEFINED CMAKE_INSTALL_RPATH)
+    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+endif()
+
 set_target_properties(${TARGET}
     PROPERTIES
         EXPORT_COMPILE_COMMANDS ON
         RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
-        INSTALL_RPATH            "${CMAKE_INSTALL_PREFIX}/lib"
+        INSTALL_RPATH            "${CMAKE_INSTALL_RPATH}"
 )
diff -ruN whispercpp-1.7.1-org/CMakeLists.txt whispercpp-1.7.1-patched/CMakeLists.txt
--- whispercpp-1.7.1-org/CMakeLists.txt	2024-10-07 12:06:48.000000000 +0200
+++ whispercpp-1.7.1-patched/CMakeLists.txt	2024-10-26 16:20:21.457552399 +0200
@@ -143,8 +143,12 @@
 
 get_directory_property(WHISPER_TRANSIENT_DEFINES COMPILE_DEFINITIONS)
 
-set_target_properties(whisper PROPERTIES PUBLIC_HEADER ${CMAKE_CURRENT_SOURCE_DIR}/include/whisper.h)
-install(TARGETS whisper LIBRARY PUBLIC_HEADER)
+if (NOT DEFINED WHISPER_TARGET_NAME)
+    set(WHISPER_TARGET_NAME whisper)
+endif()
+
+set_target_properties(${WHISPER_TARGET_NAME} PROPERTIES PUBLIC_HEADER ${CMAKE_CURRENT_SOURCE_DIR}/include/whisper.h)
+install(TARGETS ${WHISPER_TARGET_NAME} LIBRARY PUBLIC_HEADER)
 
 configure_package_config_file(
         ${CMAKE_CURRENT_SOURCE_DIR}/cmake/whisper-config.cmake.in
diff -ruN whispercpp-1.7.1-org/ggml/src/CMakeLists.txt whispercpp-1.7.1-patched/ggml/src/CMakeLists.txt
--- whispercpp-1.7.1-org/ggml/src/CMakeLists.txt	2024-10-07 12:06:48.000000000 +0200
+++ whispercpp-1.7.1-patched/ggml/src/CMakeLists.txt	2024-10-26 19:54:50.084636258 +0200
@@ -254,6 +254,9 @@
         "https://cmake.org/cmake/help/latest/module/FindBLAS.html#blas-lapack-vendors"
         " to set correct GGML_BLAS_VENDOR")
     endif()
+
+
+
 endif()
 
 if (GGML_LLAMAFILE)
@@ -412,9 +415,15 @@
         set(ROCM_PATH $ENV{ROCM_PATH})
     endif()
 
-    list(APPEND CMAKE_PREFIX_PATH  ${ROCM_PATH})
+    set(ROCM_PATH /opt/rocm CACHE STRING "ROCm root dir")
+    list(APPEND CMAKE_PREFIX_PATH ${ROCM_PATH})
     list(APPEND CMAKE_PREFIX_PATH "${ROCM_PATH}/lib64/cmake")
 
+    message(STATUS "using ROCm root path ${ROCM_PATH}")
+
+    set(CMAKE_C_COMPILER "${ROCM_PATH}/llvm/bin/clang")
+    set(CMAKE_CXX_COMPILER "${ROCM_PATH}/llvm/bin/clang++")
+
     # CMake on Windows doesn't support the HIP language yet
     if (WIN32)
         set(CXX_IS_HIPCC TRUE)
@@ -1319,7 +1328,7 @@
 
 # ggml
 
-add_library(ggml
+add_library(ggml STATIC
             ../include/ggml.h
             ../include/ggml-alloc.h
             ../include/ggml-backend.h
diff -ruN whispercpp-1.7.1-org/src/CMakeLists.txt whispercpp-1.7.1-patched/src/CMakeLists.txt
--- whispercpp-1.7.1-org/src/CMakeLists.txt	2024-10-07 12:06:48.000000000 +0200
+++ whispercpp-1.7.1-patched/src/CMakeLists.txt	2024-10-26 18:19:08.745167663 +0200
@@ -80,39 +80,40 @@
 
 # whisper
 
-add_library(whisper
+add_library(${WHISPER_TARGET_NAME}
             ../include/whisper.h
             whisper.cpp
             )
 
 # Set the version numbers
-set_target_properties(whisper PROPERTIES
+set_target_properties(${WHISPER_TARGET_NAME} PROPERTIES
     VERSION ${PROJECT_VERSION}
     SOVERSION ${SOVERSION}
 )
 
-target_include_directories(whisper PUBLIC . ../include)
-target_compile_features   (whisper PUBLIC cxx_std_11) # don't bump
+target_include_directories(${WHISPER_TARGET_NAME} PUBLIC . ../include)
+target_include_directories(${WHISPER_TARGET_NAME} PUBLIC . ../ggml/include)
+target_compile_features   (${WHISPER_TARGET_NAME} PUBLIC cxx_std_11) # don't bump
 
 if (WHISPER_EXTRA_FLAGS)
-    target_compile_options(whisper PRIVATE ${WHISPER_EXTRA_FLAGS})
+    target_compile_options(${WHISPER_TARGET_NAME} PRIVATE ${WHISPER_EXTRA_FLAGS})
 endif()
 
-target_link_libraries(whisper PUBLIC ggml)
+target_link_libraries(${WHISPER_TARGET_NAME} PUBLIC ggml)
 
 if (WHISPER_COREML)
-    target_link_libraries(whisper PRIVATE whisper.coreml)
+    target_link_libraries(${WHISPER_TARGET_NAME} PRIVATE whisper.coreml)
 endif()
 
 if (WHISPER_OPENVINO)
-    target_link_libraries(whisper PRIVATE whisper.openvino)
+    target_link_libraries(${WHISPER_TARGET_NAME} PRIVATE whisper.openvino)
 endif()
 
 if (WHISPER_MKL)
-    target_link_libraries(whisper PRIVATE MKL::MKL)
+    target_link_libraries(${WHISPER_TARGET_NAME} PRIVATE MKL::MKL)
 endif()
 
 if (BUILD_SHARED_LIBS)
-    set_target_properties(whisper PROPERTIES POSITION_INDEPENDENT_CODE ON)
-    target_compile_definitions(whisper PRIVATE WHISPER_SHARED WHISPER_BUILD)
+    set_target_properties(${WHISPER_TARGET_NAME} PROPERTIES POSITION_INDEPENDENT_CODE ON)
+    target_compile_definitions(${WHISPER_TARGET_NAME} PRIVATE WHISPER_SHARED WHISPER_BUILD)
 endif()
